File: Microsoft.Toolkit.Uwp.Input.GazeInteraction/InvokePatternGazeTargetItem.cpp
#include "pch.h"
#include "InvokePatternGazeTargetItem.h"
#include "InvokePatternGazeTargetItem.g.cpp"
#include "winrt/Microsoft.UI.Xaml.Automation.Peers.h"
#include "winrt/Microsoft.UI.Xaml.Automation.Provider.h"

using namespace winrt;
using namespace winrt::Microsoft::UI::Xaml::Automation::Peers;
using namespace winrt::Microsoft::UI::Xaml::Automation::Provider;

namespace winrt::Microsoft::Toolkit::Uwp::Input::GazeInteraction::implementation
{
    void InvokePatternGazeTargetItem::Invoke()
    {
        auto peer = FrameworkElementAutomationPeer::FromElement(TargetElement());
        auto provider = peer.GetPattern(PatternInterface::Invoke).try_as<IInvokeProvider>();
        provider.Invoke();
    }
}

File: Microsoft.Toolkit.Uwp.Input.GazeInteraction/InvokePatternGazeTargetItem.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using Windows.UI.Xaml;
using Windows.UI.Xaml.Automation.Peers;
using Windows.UI.Xaml.Automation.Provider;

namespace Microsoft.Toolkit.Uwp.Input.GazeInteraction
{
    internal class InvokePatternGazeTargetItem : GazeTargetItem
    {
        internal InvokePatternGazeTargetItem(UIElement element)
            : base(element)
        {
        }

        internal override void Invoke()
        {
            var peer = FrameworkElementAutomationPeer.FromElement(TargetElement);
            var provider = peer.GetPattern(PatternInterface.Invoke) as IInvokeProvider;
            provider.Invoke();
        }
    }
}